internal/app/referenceserver/server.go

```diff
@@ -111,6 +111,9 @@ type stdHTTPServer struct {
 }
 
 func (s *stdHTTPServer) Serve() error {
+	if s.svr.TLSConfig != nil {
+		return s.svr.ServeTLS(s.lis, "", "")
+	}
```

Comment:
Doh! This is necessary for the HTTP server to actually be a secure server. The empty strings are for the cert and key, but they can be blank if they are instead provided in the server's `TLSConfig` field (which they are in this case).